docs/src/examples.rst

```diff
@@ -14,205 +14,216 @@
 
-The following example creates the |S3| bucket **MyS3Bucket** with server-side KMS
-encryption provided by |S3|.
+There are two different ways of running your container tasks with |ECS|:
 
-.. code-block:: js
+- Using the **Fargate** launch type, where |ECS| manages the physical machines that your containers are running on for you
+- Using the **EC2** launch type, where you do the managing, such as specifying autoscaling
 
-   import s3 = require('@aws-cdk/aws-s3');
-   import cdk = require('@aws-cdk/cdk');
+This example creates a Fargate service,
+which requires a VPC, a cluster, and a task definition.
 
-   class MyStack extends cdk.Stack {
-     constructor(parent: cdk.App, name: string, props?: cdk.StackProps) {
-       super(parent, name, props);
+.. tabs::
 
-       new s3.Bucket(this, 'MyBucket', {
-         bucketName: 'MyS3Bucket',
-         encryption: s3.BucketEncryption.KmsManaged
-       });
-     }
-   }
+    .. group-tab:: TypeScript
 
-   const app = new cdk.App();
+        Add the following import statements:
 
-   new MyStack(app, 'MyStack');
+        .. code-block:: typescript
 
-   app.run()
+            import ec2 = require('@aws-cdk/aws-ec2');
+            import ecs = require('@aws-cdk/aws-ecs');
+
+        Add the following code to the end of the constructor:
+
+        .. code-block:: typescript
+
+            const vpc = new ec2.VpcNetwork(this, 'MyVpc', {
+              maxAZs: 3 // Default is all AZs in region
+            });
+
+            // Create an ECS cluster
+            const cluster = new ecs.Cluster(this, 'MyCluster', {
+              vpc: vpc
+            });
+
+            const taskDefinition = new ecs.FargateTaskDefinition(this, 'MyFargateTaskDefinition', {
+              cpu: '512',  // Default is 256
+              memoryMiB: '2048'  // Default is 512
+            });
+
+            // The task definition for the container.
+            taskDefinition.addContainer('MyContainer', {
+              image: ecs.ContainerImage.fromDockerHub('amazon/amazon-ecs-sample')    // Required
+            });
+
+            new ecs.FargateService(this, 'MyFargateService', {
+              taskDefinition: taskDefinition,  // Required
+              cluster: cluster,  // Required
+              desiredCount: 6,  // Default is 1
+            });
+
+        Save it and make sure it builds and creates a stack.
+
+        .. code-block:: sh
+
+            npm run build
+            cdk synth
+
+        You should see a stack of a couple hundred lines, so we won't show it here.
+        The stack should contain one default instance, a private subnet and a public subnet
+        for the three availability zones, and a security group.
+
+        Deploy the stack.
+
+        .. code-block:: sh
+
+            cdk deploy
+
+        |CFN| displays information about the dozens of steps that
+        it takes as it deploys your app.
+
+That's how easy it is to create a Fargate service to run a Docker image.
```

Comment:
It's true. The example is not yet useful without a load balancer though, since this is a web service but you will be unable to hit the endpoints (because they'll be in a private VPC if I'm not mistaken).

Just sayin'.

Reply:
Are you planning to make this section longer, or is it the plan to publish like this?

Reply:
I'll add a load balancer and auto-scaling for tasks to the example. Once the new template is released, I'll fix the references:

*my_ecs_construct.ts* in the *bin* directory -> *my_ecs_stack.ts* in the *lib* directory